src/components/SlidingNavigation/SlidingNavigation.js

```diff
@@ -0,0 +1,112 @@
+import * as React from "react";
+import PropTypes from "prop-types";
+import classNames from "classnames";
+
+import "./SlidingNavigation.css";
+
+const SlidingNavigation = ({ data, options }) => {
+  const activeItemClassName = `SlidingNavigation__item--active`;
+  const observerOptions = options || {
+    threshold: 0.2,
+  };
+
+  React.useEffect(() => {
+    const observer = new IntersectionObserver((entries) => {
+      entries.forEach((entry) => {
+        if (
+          entry.isIntersecting &&
+          !window.location.hash.includes(entry.target.id)
+        ) {
+          window.history.replaceState({}, ``, `#${entry.target.id}`);
+          const currentActiveElement = document.querySelector(
```

Comment:
In this case use reference (useRef)

Reply:
I'm not sure how can I put a `ref={activeRef}` on the current active element, since they're mapped out, so the ref would be on all 3 elements, or not?

Reply:
```js
const navigationItemRef = React.useRef(null)
console.log({ navigationItemRef });

/* rest of code */

<div
  ref={navigationItemRef}
  id={`${item.linkId}-sn`}
  className="SlidingNavigation__item""
  onClick={...}
  key={item.linkId}
>
  {!!item.icon && (
    <span
      className={`SlidingNavigation__icon SlidingNavigation__icon--${item.icon}`}
    />
  )}
    {item.title}
  </div>
```

Putting a ref on the mapped-out div results in the console.log returning me a ref of the **last** map item:
![image](https://user-images.githubusercontent.com/69549795/168066908-2a4d3ade-8829-44e6-98b8-f6ab454aab85.png)

---
Granted, I'm not the sharpest tool in the shelf when it comes to `useRef` so I might be missing something obvious @bmarty92

Reply:
https://mattclaffey.medium.com/adding-react-refs-to-an-array-of-items-96e9a12ab40c there are lot of info sources, for example this.